e2e/app/page/cohort-participants-group.ts

```diff
@@ -513,17 +514,29 @@ export default class CohortParticipantsGroup {
   }
 
   async searchCriteria(searchWord: string): Promise<Table> {
-    const searchFilterTextbox = Textbox.findByName(this.page, { dataTestId: 'list-search-input' });
-    const waitForResponsePromise = this.page.waitForResponse(
-      (response) => {
-        return response.url().includes('/criteria/') && response.request().method() === 'GET';
-      },
-      { timeout: 60000 }
-    );
-    await searchFilterTextbox.type(searchWord);
-    await searchFilterTextbox.pressReturn();
-    await waitForResponsePromise;
-    await waitWhileLoading(this.page);
+    const searchTextbox = Textbox.findByName(this.page, { dataTestId: 'list-search-input' });
+
+    const enterTextAndSearch = async (searchString: string): Promise<boolean> => {
+      const responsePromise = this.page.waitForResponse(
+        (response) => {
+          return response.url().includes('/criteria/') && response.request().method() === 'GET';
+        },
+        { timeout: 60000 }
+      );
+      await searchTextbox.type(searchString);
+      await searchTextbox.pressReturn();
+      const response = await responsePromise;
+      await waitWhileLoading(this.page);
+      return response.ok();
+    };
+
+    const isSuccess = await enterTextAndSearch(searchWord);
+    if (!isSuccess) {
+      logger.info('RETRY search criteria in 5 seconds after API request failure.');
+      await this.page.waitForTimeout(5000);
+      await enterTextAndSearch(searchWord);
+    }
+
```

Comment:
Refactor `searchCriteria` function to make it resilient - Adds retry because api request sometimes fails. Hopefully request works the second time.

<img width="1354" alt="Screen Shot 2022-01-20 at 10 47 46 PM" src="https://user-images.githubusercontent.com/35533885/150462446-db5c8bc2-842d-44c4-8185-f6f3cc64e8f1.png">